### PR TITLE
Fix compiler warnings issue found in conformance_test_runner#8189

### PR DESCRIPTION
**Pull request by akjain2052**
Signed-off-by: Abhishek Jain <abhishek.jain3@huawei.com>

### CODE REVIEW

**Comment by akjain2052** [comment]
This is a fix for this issue - **[C++] Compiler warnings showing in conformance_test_runner #8189**

**Comment by akjain2052** [comment]
Please add label:c++ and release notes:yes

File: conformance/conformance_test_runner.cc [hunk 1]
```diff
@@ -307,7 +307,8 @@ void ForkPipeRunner::SpawnTestProgram() {
 }
 
 void ForkPipeRunner::CheckedWrite(int fd, const void *buf, size_t len) {
-  if (write(fd, buf, len) != len) {
```
Comment (TeBoring) [inline conformance/conformance_test_runner.cc:310]: static_cast?

Comment (akjain2052) [reply conformance/conformance_test_runner.cc:310]: @TeBoring , I have updated the change as per your comments. Please review.

**Comment by akjain2052** [comment]
Please review the changes, and help to add label:c++ & release notes:yes

**Comment by akjain2052** [comment]
@TeBoring @acozzette Please review the changes, and help to add label:c++ & release notes:yes

File: conformance/conformance_test_runner.cc (hunk 1 above)
Comment (TeBoring) [reply conformance/conformance_test_runner.cc:310]: should this be `static_cast<size_t>`?

Comment (akjain2052) [reply conformance/conformance_test_runner.cc:310]: Yes. Updated the changes. Thanks for review.

**Comment by akjain2052** [comment]
Please help to review, and merge this change.

File: conformance/conformance_test_runner.cc (hunk 1 above)
Comment (akjain2052) [reply conformance/conformance_test_runner.cc:310]: @TeBoring Please review the change, and help to merge this

**Review by dlj-NaN — APPROVED** [review]

**Pull request merged**